app/views/Settings.js

```diff
@@ -55,6 +52,41 @@ export const SettingsScreen = ({ navigation }) => {
     };
   }, [navigation]);
 
+  // checking whether location permission is enable or disbale from application setting on load of setting's page.
+  const checkIsLocationEnable = async () => {
+    await LocationServices.checkStatus()
+      .then(res => {
+        if (res !== undefined) {
+          if (res.isRunning) {
+            SetStoreData(PARTICIPATE, true);
+            setIsLogging(true);
+          }
+        }
+      })
+      .catch(err => {
+        console.log('Is background Location enable ', err);
+      });
+  };
+
+  //  This method is used to track whether user has "Allow" or "Deny" location permission system Dialog.
+  const systemDialogLocationPermission = async () => {
+    await LocationServices.getSystemLocationPermission()
+      .then(res => {
```

Comment:
This isn't quite right - I think taking a look at this link may be helpful https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

The short answer is that `await` does not require a `.then()`.